app/services/case_updater_service.rb

```diff
@@ -1,35 +1,61 @@
 class CaseUpdaterService
-  attr_reader :result
+  attr_reader :result, :error_message
 
   def initialize(user, kase, params)
     @user = user
     @team = kase.managing_team
     @kase = kase
     @params = params
     @result = :incomplete
+    @error_message = nil
   end
 
   def call
-    ActiveRecord::Base.transaction do
-      begin
+    begin
+      ActiveRecord::Base.transaction do
+        # There is extra checking for linked cases as those changes won't appear in the 
+        # case's changed_attributes, has to be checked separately
+        linked_cases_changed = has_linked_cases_changed?
         @kase.assign_attributes(@params)
         # properties is JSON object, if one of keys was not included 
         # but later this key with value of nil is added, 
         # it will be treated as changed which will give a misleading message 
         # as from user's point of view, he/she doesn't change anything
         # Each key within this JSON object will be tracked individually, 
         # no need for tracking properties as the whole.
-        if (@kase.changed_attributes.keys - ["properties"]).present?
+        if (@kase.changed_attributes.keys - ["properties"]).present? || linked_cases_changed
           @kase.save!
           @kase.state_machine.edit_case!(acting_user: @user, acting_team: @team)
           @result = :ok
         else
           @result = :no_changes
         end
-      rescue
-        @result = :error
       end
-
+    rescue StandardError => error
+      @error_message = error.message
+      @result = :error
     end
   end
+
+  private
+
+  def has_linked_cases_changed?
+    return has_original_case_changed? || has_related_cases_changed?
+  end
+
+  def has_original_case_changed?
+    return (@kase.respond_to? :original_case_ids) && 
+            @params[:original_case_ids].present? && 
+            !two_arrays_is_equal?(@kase.original_case_ids, @params[:original_case_ids])
+  end
+
+  def has_related_cases_changed?
+    return (@kase.respond_to? :related_case_ids) &&
+            @params[:related_case_ids].present? && 
+            !two_arrays_is_equal?(@kase.related_case_ids, @params[:related_case_ids])
+  end
+
+  def two_arrays_is_equal?(array1, array2)
```

Comment:
this might read better as `two_arrays_are_equal?` as is is singular and this refers to two things 👍